model/task/task.go

```diff
@@ -249,6 +249,35 @@ type TestResult struct {
 	LogRaw string `json:"log_raw" bson:"log_raw,omitempty"`
 }
 
+type DisplayTaskCache struct {
+	execToDisplay map[string]*Task
+	displayTasks  []*Task
+}
+
+func (c *DisplayTaskCache) Get(t *Task) (*Task, error) {
+	if parent, exists := c.execToDisplay[t.Id]; exists {
+		return parent, nil
+	}
+	displayTask, err := t.GetDisplayTask()
```

Comment:
I don't think we'll encounter this here, but I think there's a bug in GetDisplayTask():

https://github.com/evergreen-ci/evergreen/blob/8ad69cc9b75c8fa742fd6fb0b67fbd7a69e8c4fc/model/task/task.go#L1448

will return the actual task instead of the display task.

Reply:
That's a performance optimization to avoid making a query if it's been retrieved already

Reply:
I mean I think we needs to return t.DisplayTask instead of t

Reply:
Oh fixed